src/main/java/org/broadinstitute/hellbender/tools/spark/sv/BreakpointPair.java

```diff
@@ -0,0 +1,476 @@
+        }
+    }
+
+    /**
+     * @return Micro-homologyForwardStrandRep sequence using two alignments of the same contig: as indicated by their overlap on the contig itself.
+     *          Empty if they don't overlap on the contig.
+     */
+    @VisibleForTesting
+    static String getHomology(final AlignmentRegion current, final AlignmentRegion next, final byte[] contigSequence) {
+
+        if (current.endInAssembledContig >= next.startInAssembledContig) {
+            final byte[] homologyBytes = Arrays.copyOfRange(contigSequence, next.startInAssembledContig-1, current.endInAssembledContig);
+            if (current.referenceInterval.getStart() > next.referenceInterval.getStart()) {
+                SequenceUtil.reverseComplement(homologyBytes, 0, homologyBytes.length);
+            }
+            return new String(homologyBytes);
+        } else {
+            return EMPTY_STRING;
+        }
+    }
+
+    /**
+     * Note: not suitable for the most complicated case dealt with in {@link #resolveComplications(ChimericAlignment)}
+     * @return Inserted sequence using two alignments of the same contig: as indicated by their separation on the the contig itself.
+     */
+    @VisibleForTesting
+    static String getInsertedSequence(final AlignmentRegion current, final AlignmentRegion next, final byte[] contigSequence) {
+
+        if (current.endInAssembledContig < next.startInAssembledContig-1) {
+            final byte[] insertedSequenceBytes = Arrays.copyOfRange(contigSequence, current.endInAssembledContig, next.startInAssembledContig-1);
+            if (current.referenceInterval.getStart() > next.referenceInterval.getStart()) {
+                SequenceUtil.reverseComplement(insertedSequenceBytes, 0, insertedSequenceBytes.length);
+            }
+            return new String(insertedSequenceBytes);
+        } else {
+            return EMPTY_STRING;
+        }
+    }
+
+    // TODO: 12/11/16 More robust way and ways to test (currently covered, but NOT DIRECTLY)
+    /**
+     * This function, when given overlaps of two corresponding regions on reference and contig sequences,
+     * attempts to find--naively and slowly--the copy numbers on the reference and the contig of tandem repeats,
+     * as well as the pseudo-homologyForwardStrandRep between the duplicated sequence and the right flanking region.
+     * The returned array of length 4 have entries in the following order:
+     *  estimate of copy number on the sequence where {@code overlapOnHigherCNSequence} was computed,
+     *  estimate of copy number on the sequence where {@code overlapOnLowerCNSequence} was computed,
+     *  duplicated unit length,
+     *  pseudo-homologyForwardStrandRep length
+     */
+    @VisibleForTesting
+    static int[] tandemDuplicationComplication(final int overlapOnHigherCNSequence, final int overlapOnLowerCNSequence) {
+
+        int higherCnEst=0, lowerCnEst=0, unitLen=0, pseudoHomLen=0;
```

Comment:
In the inner most loop in the if statement, `unitLen` which is the length of the repeated sequence, is updated.